EventFilter/CTPPSRawToDigi/src/RawDataUnpacker.cc

```diff
@@ -80,11 +80,14 @@ int RawDataUnpacker::ProcessOptoRxFrame(const word *buf, unsigned int frameSize,
   #endif
 
   // parallel or serial transmission?
-  if (FOV == 1)
-    return ProcessOptoRxFrameSerial(buf, frameSize, fc);
-
-  if (FOV == 2)
-    return ProcessOptoRxFrameParallel(buf, frameSize, fedInfo, fc);
+  switch (FOV) {
+    case 1:
+      return ProcessOptoRxFrameSerial(buf, frameSize, fc);
+    case 2:
+    case 3:
```

Comment:
Exactly: it is an extension of the FOV==2, including the compact data format for the diamonds.
Since it is anyway a new firmware, the FOV was incremented.